frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -15,157 +15,215 @@
+  const [tooltipData, setTooltipData] = useState<{
+    toolTipVisible: boolean;
+    toolTipValue?: string;
+    toolTipTop: number;
+    toolTipLeft: number;
+    toolTipTime: number;
+  }>({
     toolTipVisible: false,
-    toolTipValue: undefined,
     toolTipTop: 0,
     toolTipLeft: 0,
     toolTipTime: 0,
-    isMobile: false,
-  };
+  });
 
-  public componentDidMount() {
-    if (this.state.source !== 'hourly' && this.context.chartDisplay === 'week') {
-      this.setState({ source: 'hourly' });
-    }
-    this.createChart();
+  const prevChartDataDaily = usePrevious(data.chartDataDaily);
+  const prevChartDataHourly = usePrevious(data.chartDataHourly);
+  const prevChartFiat = usePrevious(data.chartFiat);
+  const prevHideAmounts = usePrevious(hideAmounts);
+
+  if (source === 'daily' && prevChartDataDaily?.length !== data.chartDataDaily.length) {
+    lineSeries.current?.setData(data.chartDataDaily);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataDaily);
   }
 
-  public componentWillUnmount() {
-    window.removeEventListener('resize', this.onResize);
-    if (this.chart) {
-      this.chart.timeScale().unsubscribeVisibleLogicalRangeChange(this.calculateChange);
-      this.chart.unsubscribeCrosshairMove(this.handleCrosshair);
-    }
+  if (source === 'hourly' && prevChartDataHourly?.length !== data.chartDataHourly.length) {
+    lineSeries.current?.setData(data.chartDataHourly);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataHourly);
   }
 
-  public componentDidUpdate(prev: Props) {
-    const { chartDataDaily, chartDataHourly } = this.props.data;
-    if (!this.chart) {
-      this.createChart();
-    }
-    if (
-      (this.lineSeries && prev.data.chartDataDaily && prev.data.chartDataHourly && chartDataDaily && chartDataHourly)
-        && (
-          prev.data.chartDataDaily.length !== chartDataDaily.length
-          || prev.data.chartDataHourly.length !== chartDataHourly.length
-        )
-    ) {
-      const data = this.state.source === 'hourly' ? chartDataHourly : chartDataDaily;
-      this.lineSeries.setData(data);
-      this.chart?.timeScale().fitContent();
-      this.setFormattedData(data);
+  if (prevChartFiat !== data.chartFiat) {
+    reinitializeChart();
+  }
+
+  if (prevHideAmounts !== hideAmounts) {
+    chart.current?.applyOptions({
+      leftPriceScale: {
+        visible: hideAmounts ? false : !isMobile,
+      }
+    });
+  }
+
+  useEffect(() => {
+    if (!chart.current) {
+      initChart();
     }
+  });
 
-    if (prev.hideAmounts !== this.props.hideAmounts) {
-      this.chart?.applyOptions({
-        leftPriceScale: {
-          visible: this.props.hideAmounts ? false : !this.state.isMobile,
-        }
-      });
+  useEffect(() => {
+    if (source !== 'hourly' && chartDisplay === 'week') {
+      setSource('hourly');
     }
+    initChart();
+    return () => {
+      window.removeEventListener('resize', onResize);
```

Comment:
initChart conditionally adds the event listener, but here it always removes the event.

I think this should be changed somehow. I guess the event can always be added?